common/flatpak-transaction.c

```diff
@@ -2879,6 +2900,89 @@ flatpak_transaction_run (FlatpakTransaction *transaction,
   return FLATPAK_TRANSACTION_GET_CLASS (transaction)->run (transaction, cancellable, error);
 }
 
+static gboolean
+get_installation_dir_free_space (FlatpakInstallation  *installation,
+                                 guint64              *free_space,
+                                 GError              **error)
+{
+  g_autoptr (GFile) installation_dir = NULL;
+  g_autoptr (GFileInfo) info = NULL;
+
+  installation_dir = flatpak_installation_get_path (installation);
+
+  info = g_file_query_filesystem_info (installation_dir,
+                                       G_FILE_ATTRIBUTE_FILESYSTEM_FREE,
+                                       NULL,
+                                       error);
+  if (info == NULL)
+    return FALSE;
+
+  *free_space = g_file_info_get_attribute_uint64 (info, G_FILE_ATTRIBUTE_FILESYSTEM_FREE);
+  return TRUE;
+}
+
+static gboolean
+check_for_disk_space (FlatpakTransaction *self)
```

Comment:
Would be tidier to take a `GError **` argument here, and do any warnings at a higher level (or propagate the `GError` even further), rather than tying any code which calls this function into potentially emitting runtime warnings.